server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/ObjectStorageBlobsDAOBuilder.java

```diff
@@ -0,0 +1,49 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.blob.objectstorage;
+
+import org.apache.james.blob.api.BlobId;
+import org.jclouds.blobstore.BlobStore;
+
+import com.google.common.base.Preconditions;
+
+public abstract class ObjectStorageBlobsDAOBuilder {
+    private ContainerName containerName;
+    private BlobId.Factory blobIdFactory;
```

Comment:
We tend to use `Optional` here

Reply:
I am not sure I understand what you mean ...

Reply:
I used `Optional` for optional parameters when it made sense, but these are mandatory fields in a mutable builder design.